srcalc/src/main/java/gov/va/med/srcalc/domain/variable/NumericalRange.java

```diff
@@ -0,0 +1,253 @@
+     */
+    public static final float MIN = -MAX;
+    
+    private static final Logger fLogger = LoggerFactory.getLogger(NumericalRange.class);
+
+    private float fLowerBound;
+    private boolean fLowerInclusive;
+    private float fUpperBound;
+    private boolean fUpperInclusive;
+    
+    /**
+     * For reflection-based construction only. Business code should use
+     * {@link #NumericalRange(float, boolean, float, boolean)}.
+     */
+    NumericalRange()
+    {
+    }
+    
+    /**
+     * Constructs an instance.
+     * @throws IllegalArgumentException if NaN is given for the lower or upper
+     * bound.
+     */
+    public NumericalRange(
+            float lowerBound,
+            boolean lowerInclusive,
+            float upperBound,
+            boolean upperInclusive)
+    {
+        // Use bounds setters to validate the arguments
+        setLowerBound(lowerBound);
+        fLowerInclusive = lowerInclusive;
+        setUpperBound(upperBound);
+        fUpperInclusive = upperInclusive;
+    }
+    
+    public float getLowerBound()
+    {
+        return fLowerBound;
+    }
+    
+    /**
+     * For reflection-based construction only. Sets the lower bound. Will limit
+     * the given bound per the range specified in {@link #limit(float)}.
+     * @throws IllegalArgumentException if NaN is given
+     */
+    void setLowerBound(final float lowerBound)
+    {
+        if (Float.isNaN(lowerBound))
+        {
+            throw new IllegalArgumentException("NaN is not a legal lower bound");
+        }
+
+        fLowerBound = limit(lowerBound);
+    }
+    
+    public boolean isLowerInclusive()
+    {
+        return fLowerInclusive;
+    }
+    
+    /**
+     * For reflection-based construction only.
+     */
+    void setLowerInclusive(boolean lowerInclusive)
+    {
+        fLowerInclusive = lowerInclusive;
+    }
+    
+    public float getUpperBound()
+    {
+        return fUpperBound;
+    }
+    
+    /**
+     * For reflection-based construction only. Sets the upper bound. Will limit
+     * the given bound per the range specified in {@link #limit(float)}.
+     * @throws IllegalArgumentException if NaN is given
+     */
+    void setUpperBound(final float upperBound)
+    {
+        if (Float.isNaN(upperBound))
+        {
+            throw new IllegalArgumentException("NaN is not a legal upper bound");
+        }
+
+        fUpperBound = limit(upperBound);
+    }
+    
+    public boolean isUpperInclusive()
+    {
+        return fUpperInclusive;
+    }
+    
+    /**
+     * For reflection-based construction only.
+     */
+    void setUpperInclusive(boolean upperInclusive)
+    {
+        fUpperInclusive = upperInclusive;
+    }
+    
+    /**
+     * Limits the given value to be within [-MIN,MAX]. If f > MAX, returns MAX.
+     * If f < MIN, returns MIN.
+     * @param f the value to limit
+     * @return the limited value
+     */
+    public static float limit(final float f)
```

Comment:
Since you check for NaN twice, why not move it into the limit() function instead?